src/classes/CRLP_ApiService_TEST.cls

```diff
@@ -98,84 +98,142 @@ private class CRLP_ApiService_TEST {
+            service.executeRollups(null, rollupType, rollupDefs, rollupData, false);
+            System.assert(false, 'An exception should have been thrown for invalid parameters');
+        } catch (Exception ex) {
+            System.assert(ex.getMessage().containsIgnoreCase(CallableApiParameters.PARAM_ROLLUP_PARENTID),
+                    'Message should contain "' + CallableApiParameters.PARAM_ROLLUP_PARENTID + '" param: ' + ex.getMessage());
+        }
 
-            CMT_UnitTestData_TEST.createRollupRecord('Count Donations All Year', filterGroupId,
-                    CMT_UnitTestData_TEST.RollupRecordType.AccountSoftCreditToAccount,
-                    'npo02__NumberOfClosedOpps__c', CRLP_Operation.RollupType.Count, 'Amount') + ',' +
+        // Should fail for an invalid parentId
+        try {
+            service.executeRollups(parentId, rollupType, rollupDefs, rollupData, false);
+            System.assert(false, 'An exception should have been thrown for invalid parameters');
+        } catch (Exception ex) {
+            System.assert(ex.getMessage().containsIgnoreCase(CallableApiParameters.PARAM_ROLLUP_PARENTID),
+                    'Message should contain "' + CallableApiParameters.PARAM_ROLLUP_PARENTID + '" param: ' + ex.getMessage());
+        }
+    }
 
-            // Contact Soft Credit
-            CMT_UnitTestData_TEST.createRollupRecord('Contact: Soft Credits - Total', filterGroupId,
-                    CMT_UnitTestData_TEST.RollupRecordType.SoftCreditToContact,
-                    'npo02__Soft_Credit_Total__c', CRLP_Operation.RollupType.Sum, 'Amount__c') + ',' +
+    /**
+     * @description Verify that the call to executeRollups() successfully completes a rollup and that
+     * the target (summary) record is not updated as a result.
+     */
+    @IsTest
+    private static void shouldCompleteExecuteRollupsForAccount() {
+        UTIL_CustomSettingsFacade.getRollupSettingsForTests(new Customizable_Rollup_Settings__c (
+                Customizable_Rollups_Enabled__c = true
+        ));
 
-            CMT_UnitTestData_TEST.createRollupRecord('Count Donations All Year', filterGroupId,
-                    CMT_UnitTestData_TEST.RollupRecordType.SoftCreditToContact,
-                    'npo02__NumberOfClosedOpps__c', CRLP_Operation.RollupType.Count, 'Amount') + ',' +
+        CMT_UnitTestData_TEST.mockFullSetOfRollupDefinitions();
+
+        String rollupType = CRLP_RollupProcessingOptions.RollupType.AccountHardCredit.name();
+        Double oppAmount = 100;
+        Id oppId = UTIL_UnitTestData_TEST.mockId(Opportunity.SObjectType);
+
+        Account acct = UTIL_UnitTestData_TEST.createMultipleTestAccounts(1, null)[0];
+        insert acct;
+        acct = [SELECT npo02__NumberOfClosedOpps__c FROM Account WHERE Id = :acct.Id LIMIT 1];
+        Double acctCountOfOppsValue = acct.npo02__NumberOfClosedOpps__c;
+
+        Map<Id, Map<SObjectType, List<SObject>>> rollupData = new Map<Id, Map<SObjectType, List<SObject>>>{
```

Comment:
- Can `data` building be extracted into a test helper method?
- Let's use helper methods to query data (for example `[SELECT npo02__NumberOfClosedOpps__c FROM Account WHERE Id = :acct.Id LIMIT 1]`)